game/game/damagecalculator.cpp

```diff
@@ -51,17 +51,44 @@ DamageCalculator::Val DamageCalculator::damageFall(Npc& npc, float speed) {
   }
 
 DamageCalculator::Val DamageCalculator::rangeDamage(Npc& nsrc, Npc& nother, const Bullet& b, const CollideMask bMsk) {
-  bool invinsible = !checkDamageMask(nsrc,nother,&b);
-  if(b.pathLength() > float(MaxBowRange) * b.hitChance() && b.hitChance()<1.f)
-    return Val(0,false,invinsible);
+  float dist       = b.pathLength();
+  bool  noHit      = dist>float(MaxMagRange);
+  bool  invincible = !checkDamageMask(nsrc,nother,&b);
+  auto  dmg        = b.damage();
+
+  if(!b.isSpell()) {
+    auto& script    = nsrc.world().script();
+    int   hitChance = int(script.rand(100));
+    int   hitCh     = 0;
+    bool  g2        = Gothic::inst().version().game==2;
+    float critRange = g2 ? CritBowRange : (CritBowRange + 500);
```

Comment:
`500` is magical. Can be `ReferenceBowRangeG1`/`ReferenceBowRangeG2`.

Also this is more `Reference` range, since at 15meters hit chance matches player stats exactly.